### PR TITLE
Max Length Subarray with sum k added

### PR DESCRIPTION
**Pull request by thevivekpande**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Description
CPP solution for max length subarray with sum k added

### Checklist

- [ Yes] I've read the contribution guidelines.
- [ Yes] I've checked the issue list before deciding what to submit.
- [ Yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1813

### CODE REVIEW

File: C-Plus-Plus/cp/MaxLengthSubarrayWithSumK.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#include<bits/stdc++.h>
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/MaxLengthSubarrayWithSumK.cpp:3]: Add a beginner description to the Code

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Tiny change. Code looks good 👍 

Please make apt README changes as well

**Comment by thevivekpande** [comment]
Readme has been updated. please have a look.

File: C-Plus-Plus/cp/MaxLengthSubarrayWithSumK.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+/*
+
+----Inputs----
+Enter size of array : 6
+Enter Array : 1 2 8 1 1 2
+Enter value of k : 10
+
+----Output----
+Length of Subarray with sum 10 = 3
+
```
Comment (vatsalkesarwani12) [inline C-Plus-Plus/cp/MaxLengthSubarrayWithSumK.cpp:69]: Add time and space complexity

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/cp/MaxLengthSubarrayWithSumK.cpp [hunk 3]
```diff
@@ -0,0 +1,67 @@
+// CPP program to find maximum length subarray with sum = k.
```
Comment (atarax665) [inline C-Plus-Plus/cp/MaxLengthSubarrayWithSumK.cpp:1]: Add a line break after the problem introduction.

File: C-Plus-Plus/cp/MaxLengthSubarrayWithSumK.cpp [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+            // updation of maxLength
+            if (maxLength < (i - map[sum - k])) 
+                maxLength = i - map[sum - k]; 
+        } 
+    } 
+    return maxLength; 
+}
+
+int main(){
+    int n, arr[100], length, k;
```
Comment (atarax665) [inline C-Plus-Plus/cp/MaxLengthSubarrayWithSumK.cpp]: Declare the array after taking the size of the array as input from the user.  You don't need to declare fixed length array initially.

File: C-Plus-Plus/cp/MaxLengthSubarrayWithSumK.cpp [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+    cout<<"Enter size of array : ";
+    cin>>n;
+
+    cout<<"\nEnter Array : ";
+    for(int i=0;i<n;i++) cin>>arr[i];
```
Comment (atarax665) [inline C-Plus-Plus/cp/MaxLengthSubarrayWithSumK.cpp:44]:
```suggestion
    for(int i = 0; i < n; i++){
        cin>>arr[i];
        }
```

**Review by atarax665 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/README.md [hunk 6]
```diff
@@ -47,6 +47,13 @@
 - [Suduko Solver](cp/SudukoSolver.cpp)
 - [Target Sum triplets](cp/target_sum_triplets.cpp)
 - [Z Algorithm](cp/zalgorithm.cpp)
+- [ArraySub](cp/ARRAYSUB.cpp)
+- [Suduko Solver](cp/SudukoSolver.cpp)
+- [Josephus Problem](cp/Josephus.cpp)
+- [NQueens](cp/NQueens.cpp)
+- [Longest Subarray with Zero Sum](cp/ZeroSum.cpp)
+- [Next Largest Permutation](cp/NextPermutation.cpp)
+- [Maximum Length subarray with sum k ](cp/MaxLengthSubarrayWithSumK.cpp)
```
Comment (HarshCasper) [inline C-Plus-Plus/README.md:56]: Why are there so many README changes?

File: C-Plus-Plus/cp/MaxLengthSubarrayWithSumK.cpp [hunk 7]
```diff
@@ -0,0 +1,70 @@
+// CPP program to find maximum length subarray with sum = k.
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/MaxLengthSubarrayWithSumK.cpp:1]: Make the description more intuitive.

File: C-Plus-Plus/cp/MaxLengthSubarrayWithSumK.cpp [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    for (int i = 0; i < n; i++) { 
+        //sum=sum+arr[i];
+        sum += arr[i]; 
+
+        //if sum = k found then icrease maxLength with 1
+        if (sum == k) 
+            maxLength = i + 1; 
+        
+        //checking whether sum is present in map or not, 
+        //if sum is not present in map do enter it.
```
Comment (HarshCasper) [inline C-Plus-Plus/cp/MaxLengthSubarrayWithSumK.cpp:21]: Use Multi-Line Comments instead of Single-Line Comments

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by atarax665** [comment]
@thevivekpande Are you still working on this? Please reply to reopen this PR.

**Pull request closed without merging**